app/models/zone.rb

```diff
@@ -48,7 +55,29 @@ def find_master_server
     active_miq_servers.detect(&:is_master?)
   end
 
+  def self.create_maintenance_zone
+    begin
+      # 1) Create Maintenance zone
+      zone = new(:name        => "#{MAINTENANCE_ZONE_NAME_PREFIX}#{SecureRandom.uuid}",
+                 :description => "Maintenance Zone",
+                 :visible     => false)
+
+      zone.save!
+
+      # 2) Assign to MiqRegion
+      MiqRegion.seed if MiqRegion.my_region.nil?
+      MiqRegion.my_region.update_attributes(:maintenance_zone => zone)
+    rescue ActiveRecord::RecordInvalid
+      raise if zone.errors[:name].blank?
+      retry
+    end
+    _log.info("Creating maintenance zone...")
+    zone
+  end
+
   def self.seed
+    maintenance_zone
```

Comment:
This should call `create_maintenance_zone` which should return the maintenance zone if it already exists.

Reply:
again, it was your idea to change it

Reply:
but yes, regarding conversation ^^^